src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,100 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private int capacity = 10;
+    private int size = 0;
+    private Object[] elements = new Object[capacity];
+
     @Override
     public void add(T value) {
-
+        checkGrow();
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndexAddRemove(index);
+        checkGrow();
```

Comment:
do we need this check here?

Reply:
yes we need. If index out of range we throw exception and if size will be more than capacity we need grow our array